app/models/archived_petition.rb

```diff
@@ -0,0 +1,40 @@
+class ArchivedPetition < ActiveRecord::Base
+  OPEN_STATE = 'open'
+  REJECTED_STATE = 'rejected'
+  STATES = [OPEN_STATE, REJECTED_STATE]
+
+  validates :title, presence: true, length: { maximum: 150 }
+  validates :description, presence: true, length: { maximum: 1000 }
+  validates :state, presence: true, inclusion: STATES
+
+  searchable do
+    text :title
+    text :description
+    time :created_at, trie: true
+  end
+
+  class << self
+    def search(params)
+      query = params[:q].to_s
+      page  = [params[:page].to_i, 1].max
+
+      solr_search do
+        fulltext query
+        paginate page: page, per_page: 20
+        order_by 'created_at', 'asc'
+      end.results
+    end
+  end
+
+  def open?
+    state == OPEN_STATE && closed_at.nil?
+  end
+
+  def closed?(time = Time.current)
+    state == OPEN_STATE && !!closed_at && closed_at <= time
```

Comment:
@otlaitil normally I'd agree but if `closed_at` is `nil` then the return value of `closed?` becomes `nil` instead of `false`. I'd prefer that we returned a consistent value rather than one that varies.